File: SuperHeroAPI/Database/Entities/SuperHero.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SuperHeroAPI.Database.Entities
{
    public class SuperHero
    {
        [Key]
        public int SuperHeroID { get; set; }

        [Column( TypeName = "nvarchar(32)" )]
        public string Name { get; set; } = string.Empty;

        [Column( TypeName = "nvarchar(32)" )]
        public string FirstName { get; set; } = string.Empty;

        [Column( TypeName = "nvarchar(32)" )]
        public string LastName { get; set; } = string.Empty;

        [Column( TypeName = "nvarchar(32)" )]
        public string Place { get; set; } = string.Empty;

        [ForeignKey("Team.TeamID")]
        public int TeamID { get; set; }
        public Team Team { get; set; }

        [Column( TypeName = "smallint" )]
        public short Debut { get; set; } = 0;
    }
}

File: SuperHeroClient/src/app/_models/superHero.ts
export interface SuperHero
{
	id: number;
	name: string;
	firstName: string;
	lastName: string;
	place: string;
	debut: number;
	team: SuperHeroTeamResponse;
}

export interface SuperHeroResponse
{
	id: number;
	name: string;
	firstName: string;
	lastName: string;
	place: string;
	debut: number;
	team: SuperHeroTeamResponse;
}

export interface SuperHeroTeamResponse
{
	teamID: number;
	teamName: string;
}

export interface SuperHeroRequest
{
	id: number;
	name: string;
	firstName: string;
	lastName: string;
	place: string;
	debut: number;
	teamID: number;
}

export function SuperHero_To_SuperHeroRequest( superHero: SuperHero ): SuperHeroRequest
{
	return {
		id: superHero.id,
		name: superHero.name,
		firstName: superHero.firstName,
		lastName: superHero.lastName,
		place: superHero.place,
		debut: superHero.debut,
		teamID: superHero.team.teamID
	} as SuperHeroRequest;
}

export function SuperHeroResponse_To_SuperHeroRequest(superHeroResponse: SuperHeroResponse): SuperHeroRequest
{
	return {
		id: superHeroResponse.id,
		name: superHeroResponse.name,
		firstName: superHeroResponse.firstName,
		lastName: superHeroResponse.lastName,
		place: superHeroResponse.place,
		debut: superHeroResponse.debut,
		teamID: superHeroResponse.team.teamID
	} as SuperHeroRequest;
}

export function SuperHeroResponses_To_SuperHeroRequests(superHeroResponses: SuperHeroResponse[]): SuperHeroRequest[]
{
	let output: SuperHeroRequest[] = [];
	for (let superHeroResponse of superHeroResponses)
	{
		output.push(SuperHeroResponse_To_SuperHeroRequest(superHeroResponse));
	}
	return output;
}
